### ISSUE TITLE
Fail to detect unimplemented method in explicit Protocol subclass

### ISSUE THREAD

**Issue by shuangluoxss** [astral-sh/ty#2997]
## Description:
When a class explicitly inherits from a Protocol but not implement all required methods, the type checker should report an error about instantiating an abstract class. However, `ty` currently fails to detect this issue, while `basedpyright` correctly identifies it.
## Minimal Reproduction
```python
from typing import Protocol, runtime_checkable

@runtime_checkable
class SomeProto(Protocol):
    attr: int
    def method(self) -> str: ...

class ExplicitSubclass(SomeProto):
    def __init__(self, attr: int):
        self.attr = attr

a = ExplicitSubclass(1)  # Should error: `method` not implemented
s = a.method()  # Method doesn't exist
print(len(s))
```
<img width="940" height="878" alt="Image" src="https://github.com/user-attachments/assets/59803eb7-1c13-45bf-9472-b1b42c8caf27" />

### Version

ty 0.0.21 (c1ad9f281 2026-03-06)

**Comment by AlexWaygood** [comment]
Thanks!
